common/src/main/java/org/vivecraft/client_vr/gameplay/VRPlayer.java

```diff
@@ -476,92 +598,116 @@ public boolean getFreeMove() {
     }
 
     public String toString() {
-        return "VRPlayer: \r\n \t origin: " + this.roomOrigin + "\r\n \t rotation: " + String.format("%.3f", ClientDataHolderVR.getInstance().vrSettings.worldRotation) + "\r\n \t scale: " + String.format("%.3f", this.worldScale) + "\r\n \t room_pre " + this.vrdata_room_pre + "\r\n \t world_pre " + this.vrdata_world_pre + "\r\n \t world_post " + this.vrdata_world_post + "\r\n \t world_render " + this.vrdata_world_render;
+        return ("""
+            VRPlayer:
+                origin: %s
+                rotation: %.3f
+                scale: %.3f
+                room_pre: %s
+                world_pre: %s
+                world_post: %s
+                world_render: %s"""
+        ).formatted(
+            this.roomOrigin,
+            this.dh.vrSettings.worldRotation,
+            this.worldScale,
+            this.vrdata_room_pre,
+            this.vrdata_world_pre,
+            this.vrdata_world_post,
+            this.vrdata_world_render);
     }
 
     public Vec3 getRightClickLookOverride(Player entity, int c) {
-        Vec3 vec3 = entity.getLookAngle();
+        Vec3 out = entity.getLookAngle();
 
         if (((GameRendererExtension) this.mc.gameRenderer).vivecraft$getCrossVec() != null) {
-            vec3 = entity.getEyePosition(1.0F).subtract(((GameRendererExtension) this.mc.gameRenderer).vivecraft$getCrossVec()).normalize().reverse();
+            out = entity.getEyePosition(1.0F)
+                .subtract(((GameRendererExtension) this.mc.gameRenderer).vivecraft$getCrossVec())
+                .normalize().reverse(); // backwards
         }
 
-        ItemStack itemstack;
-        label54:
-        {
-            itemstack = c == 0 ? entity.getMainHandItem() : entity.getOffhandItem();
+        ItemStack itemStack = c == 0 ? entity.getMainHandItem() : entity.getOffhandItem();
 
-            if (!(itemstack.getItem() instanceof SnowballItem) && !(itemstack.getItem() instanceof EggItem) && !(itemstack.getItem() instanceof SpawnEggItem) && !(itemstack.getItem() instanceof PotionItem) && !(itemstack.getItem() instanceof BowItem) && !itemstack.is(ItemTags.VIVECRAFT_THROW_ITEMS)) {
-                if (!(itemstack.getItem() instanceof CrossbowItem crossbowitem)) {
-                    break label54;
-                }
+        if (itemStack.getItem() instanceof SnowballItem ||
```

Comment:
seems we forgot our tag here?

Reply:
not sure if the crossbow item tag would work there, since it actually needs to be a crossbow item for the charge check I think, unless you mean separate from the charge check

Reply:
I mean `itemstack.is(ItemTags.VIVECRAFT_THROW_ITEMS)`

Reply:
ah